### DISCUSSION TITLE
v7.7.0

### DISCUSSION THREAD

**Discussion by mvdwetering** [mvdwetering/yamaha_ynca#269] [General]
## What's Changed

* Add preset support (partially experimental)
* Add song/album info support for NetRadio (if provided by receiver)
* Add icon translation for services
* Add more supported receivers to the list in the readme, thanks for reporting 👍

## Preset support

Support for presets has been added for all inputs that seem to support it. These inputs are Napster, Netradio, Pandora, PC, Rhapsody, Sirius, SiriusIR, Tuner and USB. Tested on Netradio, Tuner and USB inputs, but others should work the same.

Presets can be selected in the mediabrowser of the mediaplayer. When selecting a preset the receiver will turn on and switch input if needed.

>**Experimental**
>Support for presets on DAB/FM tuners is experimental. That input works differently from the others and since my receiver does not have a DAB tuner it is untested. Please provide feedback if it works or not (activating and storing of presets).

Home Assistant has no standardized way to store presets, so the `store_preset` service was added. It will store a preset with the provided number for the current playing item.

Due to limitations on the protocol the integration can only show the preset number, no name or what is stored. Some presets can be managed in the Yamaha AV Control app, for others you will need to remember what was stored.

**Full Changelog**: https://github.com/mvdwetering/yamaha_ynca/compare/v7.6.0...v7.7.0

<hr /><em>This discussion was created from the release <a href='https://github.com/mvdwetering/yamaha_ynca/releases/tag/v7.7.0'>v7.7.0</a>.</em>

**Comment by tramperb** [comment]
Thank you for the integration - it's used everyday!
I'd like to select a tuner preset from an automation but don't understand what elements are needed to set up the media_player.play_media service.  What should be used for media_content_id & media_content_type?

Also, I've just noticed a small glitch with presets selected from the media browser.  When a preset is first selected, the source changes to Tuner but the preset is not set. Any preset selected after the source is changed will then work as expected (I can list this as an issue if you'd prefer).

**Reply by mvdwetering** [reply]
When building an automation you should be able to just browse for the preset and it should set the media_id under the hood. Like in the screenshot below.

![image](https://github.com/mvdwetering/yamaha_ynca/assets/732514/c8cefb64-019b-4c0f-81db-cea5ef87b621)

Hmmm...., it seems to matter how you select that service in the automation editor. When I select a new action, then scroll to "Mediaplayer" and then select "Play media" (first entry for me) it show a selection browser like in the screenshot. When I select new action and then search for mediaplayer and select the option to select play media it just shows 2 textboxes. Weird...

Format for `media_content__id` is basically "<input>:preset:<presetnumber>" except for a DAB tuner where "preset" is "fmpreset" or "dabpreset". For a normal tuner this would be "tun:preset:1" for preset 1.
For `media_content_type` use "music" (this is not really used yet, but lets keep it the same as I return from mediabrowser to avoid future breakage).
I did not document the format since I assumed it would always be possible to select with the mediabrowser, but I guess I should add it to the docs :/


For the preset not properly activating, please submit an issue with the requested diagnostics. Pretty sure it worked for me, so could be something receiver specific.

**Reply by tramperb** [reply]
> Hmmm...., it seems to matter how you select that service in the automation editor.
For me (HA in docker container), either way of selecting, I just have the text boxes as shown in the screenshot below.
![preset](https://github.com/mvdwetering/yamaha_ynca/assets/85221759/4729849e-7495-4c91-88e1-b337714b8ab7)

Finally worked it out.  I used the developer tools to listen to the event & saw that (eg for preset 3) the media_content_id should be   **tun:preset:3**    which, if I'd looked carefully is exactly what you'd written !
Thank you :)
